### ISSUE TITLE
Beat and Bassline glitch

### ISSUE THREAD

**Issue by Texxiliator** [LMMS/lmms#3544]
When BaB is played and you selelect to play some other instrument outside BaB there is the chance that at the same time note plays in BaB the instrument that plays it will lose all effects.

**Comment by Texxiliator** [comment]
Could be dangerous if you have an effect that makes instrument very quiet so you put volume up so when a note plays its EXTREMLY loud.

**Comment by musikBear** [comment]
@Texxiliator 
I think that this:
* noticed it only works with this projects / couldnt reproduce*
is the most important here.
I think that it is a project issue, and not a general lmms issue, so in that light, this is not a lmms-bug, it is a one project-specific-issue
I will take a look at you project, and try to figure out what is the reason

**Comment by musikBear** [comment]
No, it is indeed a Monstro bug
There is no need for any additional notes in song-editor. 
The issue is emphasized progressively with shortening the note in ENV.
To reproduce:
* Create an empty project
* insert default Monstro in B&B
* Set VOL of B&B to 17% (safety
* Set Monstro ENV
**D 0
** Att 10
** H 0
** D 0.15
** S 0
** R 0.1
( eg as pluck note )
* insert notes in B&B for the track either as BB or in open piano-roll
* Insert a block for the BB-track in song-editor
* start playing (use a loop

Now solo and un-solo at random. After a few of those, a loud artefact happens

**Comment by Texxiliator** [comment]
@musikBear ohh

**Comment by zonkmachine** [comment]
Bisected. This is buffer noise related to the Memory Manger, https://github.com/LMMS/lmms/pull/1088 .

**Comment by mamins1376** [comment]
It's not just with Monstro! I can reproduce it with others like Triple and BitInvader. seems the problem comes from envelope function.
- add a default BitInvader on a BaB
- make a pad-like envelope (some ATT, H=D=0, SUS=100%, some R)
- add some notes
- hit play inside piano roll
- deactivate BaB inside playlist

**Comment by zonkmachine** [comment]
Isn't this a duplicate of https://github.com/LMMS/lmms/issues/1947 which was closed in https://github.com/LMMS/lmms/pull/3804 ?
@PhysSong

**Comment by mamins1376** [comment]
It's fixed on stable-1.2 but it behaves a little odd when you mute the BnB while it's playing (from BnB, not piano-roll). actually the last note which was playing will last until it's ending. is this desired?

**Comment by zonkmachine** [comment]
> from BnB, not piano-roll). actually the last note which was playing will last until it's ending

When I mute the BBTrack in the Song Editor and there is a note on it's like turning off the envelope.

> is this desired?

Nope.

**Comment by zonkmachine** [comment]
Right. I had another go at bisecting this. As it turns out there was a related mute issue with single tracks in a bbtrack. https://github.com/LMMS/lmms/issues/1512, which was closed in https://github.com/LMMS/lmms/commit/146577d34b32ed04a426a223eb55a053d214028f by @grejppi

**Comment by PhysSong** [comment]
>it behaves a little odd when you mute the BnB while it's playing 

I can explain the reason.
Short answer: mute from track blocks sound, mute from BB blocks note on.

Long answer: `AudioPort::doProcessing()` tests whether related objects(including tracks) is muted, and block sounds if it is muted.
However, mute from BB is handled in `BBTrack::play()`. If a BB track is muted, it doesn't call subtracks' `play()` function. In such situations, it blocks adding notes for non-single-streamed instruments.

**Comment by PhysSong** [comment]
> This is buffer noise related to the Memory Manger, #1088 .

It seems to be related to #1088, but not to the memory manager. As you can see, the PR includes some changes, which seems unrelated to the memory manager.

> Isn't this a duplicate of #1947 which was closed in #3804 ?

The second one is, but the original issue isn't. It is related to what I explained above.
Muting a BB track doesn't block sound but **does** block note on and everything in `InstrumentTrack::processAudioBuffer()`. So notes which were playing before muting a BB track will keep playing without applying necessary things(effects, Env/LFO, and even volume and pan).
